sdk/notificationhubs/notification-hubs/src/api/beginSubmitNotificationHubJob.ts

```diff
@@ -38,12 +38,13 @@ export async function beginSubmitNotificationHubJob(
   const processProgressCallbacks = async (): Promise<void> =>
     progressCallbacks.forEach((h) => h(state));
   let resultPromise: Promise<NotificationHubJob> | undefined;
-  let cancelJob: (() => void) | undefined;
   const abortController = new AbortController();
   const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
 
-  const poller: SimplePollerLike<OperationState<NotificationHubJob>, NotificationHubJob> = {
-    async poll(options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+  const poller: PollerLike<OperationState<NotificationHubJob>, NotificationHubJob> = {
```

Comment:
just curious why not leverage the buildCreatePoller from core-lro directly https://github.com/Azure/azure-sdk-for-js/blob/8ddc182af9e41a25dc8e12daf2b2ba7ba50e92d8/sdk/core/core-lro/src/poller/poller.ts#L19